src/utils/QuoteParser.sol

```diff
@@ -2,40 +2,145 @@
+        report.tdAttributes = bytes8(rawReportBody.substring(120, 8));
+        report.xFAM = bytes8(rawReportBody.substring(128, 8));
+        report.mrTd = rawReportBody.substring(136, 48);
+        report.mrConfigId = rawReportBody.substring(184, 48);
+        report.mrOwner = rawReportBody.substring(232, 48);
+        report.mrOwnerConfig = rawReportBody.substring(280, 48);
+        report.rtMr0 = rawReportBody.substring(328, 48);
+        report.rtMr1 = rawReportBody.substring(376, 48);
+        report.rtMr2 = rawReportBody.substring(424, 48);
+        report.rtMr3 = rawReportBody.substring(472, 48);
+        report.reportData = rawReportBody.substring(520, 64);
     }
 
-    // Extracts and calculates the address from the public key in the TDXQuote
-    function extractAddressFromTDXQuote(TD10ReportBody calldata tdReportBody) internal pure returns (address) {
-        bytes memory publicKey = extractPublicKeyFromTDXQuote(tdReportBody);
-        return address(uint160(uint256(keccak256(publicKey))));
+    /**
+     * @notice Extracts the TEE's ephemeralEthereum address from the TD10ReportBody
+     * @dev The Ethereum address is derived using the first 64 bytes of the reportData
+     * @dev A core part of the flashtestation's protocol is that the TEE generates an ephemeral
+     * Ethereum address, which is used to identify the TEE in the AllowList. This address is derived
+     * from the TEE's public key, which is included in the quote
+     * @param td10ReportBody The TD10ReportBody to extract the Ethereum address from
+     * @return address Ethereum address
+     */
+    function extractEthereumAddress(TD10ReportBody memory td10ReportBody) internal pure returns (address) {
+        if (td10ReportBody.reportData.length != ETHEREUM_PUBLIC_KEY_LENGTH) {
+            revert InvalidReportDataLength(td10ReportBody.reportData.length);
+        }
+        return address(uint160(uint256(keccak256(td10ReportBody.reportData.substring(0, 64)))));
     }
 
-    function calculateWorkloadId(TD10ReportBody calldata tdReportBody) internal pure returns (bytes32) {
-        return keccak256(
-            abi.encodePacked(
-                tdReportBody.mrTd,
-                tdReportBody.rtMr0,
-                tdReportBody.rtMr1,
-                tdReportBody.rtMr2,
-                tdReportBody.rtMr3,
-                tdReportBody.mrOwner,
-                tdReportBody.mrOwnerConfig,
-                tdReportBody.mrConfigId
+    /**
+     * @notice Derives the TEE's workloadId from the TD10ReportBody
+     * @dev The workloadId is derived using the TEE's measurement registers
+     * @param td10ReportBody The TD10ReportBody to extract the workloadId from
+     * @return WorkloadId workloadId
+     */
+    function extractWorkloadId(TD10ReportBody memory td10ReportBody) internal pure returns (WorkloadId) {
+        return WorkloadId.wrap(
+            keccak256(
+                abi.encode(
```

Comment:
we received a request to move to abi.encodePacked. Do you see any issues with that?

Reply:
Yes I do, see my comment here: https://github.com/flashbots/rollup-boost/pull/232#discussion_r2109292999